docs/snippets/angular/your-component.ts.mdx

```diff
@@ -0,0 +1,29 @@
+```ts
+// YourComponent.stories.tsx
+import { Meta, moduleMetadata, Story } from '@storybook/angular';
+
+import { YourComponent } from './your.component';
+
+// This default export determines where your story goes in the story list
+export default {
+    title: 'YourComponent',
+    component: YourComponent,
+    decorators: [
+        moduleMetadata({
+            declarations: [
+                YourComponent,
+            ],
+        }),
+    ],
```

Comment:
This decorators section doesn't do anything as is because the `YourComponent` declaration is already handled by the Story object on the line above.

As a demonstration I think the `moduleMetadata` function is still important to include, but rather than showing an incorrect usage I would probably just add a comment to show that this is where you define anything that would normally go in the  [`@Component`](https://angular.io/api/core/Component) decorator.

Reply:
As I understood, it would act as `@NgModule` decorator …
(Since I can add `providers` and `imports` there, too).
When the component depends on another component within the same module, I'd add it to the `declarations` as well.

That being said, I haven't tried it without the `declarations` part.

Can we keep it in for consistency with other Angular examples (yet to submit)?

Reply:
Yes, sorry, you are right. `@NgModule` decorator, not the `@Component`.

You don't need the `moduleMetadata` unless the `YourComponent` also includes other dependencies. This button component example from the example folder omits it: https://github.com/storybookjs/storybook/blob/8a90cb2df6c28bfbdad9b561f24ef7799f144d49/examples/angular-cli/src/stories/doc-button/doc-button.stories.ts#L3-L7

It's definitely something a little different than with other frameworks because of the dependency injection but I'd be hesitant to include it here without a comment to explain it.

Reply:
@lachieh i'm with you on this. 
From my understanding of how Angular works and how it's implemented in Storybook per not only your example, but also for what the reader gets when he/she installs Storybook to their application it would be best to keep this example simpler and leaner. In this case we can omit it. In cases where it's required, for example when using component composition then yes, we'll need to do it. 

@Ryuno-Ki let me know when you have the changes in place so that we can get this merged.

@lachieh thank you very much for your input on this pull request, greatly appreciate it.

Reply:
Alright.
I removed the `decorators` from the Code Snippet, adjusted the indent, rebased and added it to the Setup.md file.

@jonniebigodes Can this be merged now?